lib/client/src/client/client.ts

```diff
@@ -236,31 +299,104 @@ export function createClient<R extends Range>(env: ClientEnv<R>): Client<R> {
             )
     }
 
-    const itemsChanges = (params: ItemsParams, { emitPartial }: ObserveOptions): Observable<Item[]> => {
-        return observeItems(providerClientsWithSettings(undefined), params, {
-            logger: env.logger,
-            emitPartial,
-        })
+    const filterProviders = (
+        providersObservable: Observable<ProviderClientWithSettings[]>,
+        uri?: string
+    ): Observable<ProviderClientWithSettings[]> => {
+        if (!uri) {
+            return providersObservable
+        }
+
+        return providersObservable.pipe(
+            map(providers => providers.filter(provider => provider.uri === uri))
+        )
+    }
+
+    const capabilitiesChanges = (
+        params: CapabilitiesParams,
+        { emitPartial }: ObserveOptions,
+        providerUri?: string
+    ): Observable<EachWithProviderUri<CapabilitiesResult[]>> => {
+        return observeCapabilities(
+            filterProviders(providerClientsWithSettings(undefined), providerUri),
+            params,
+            {
+                logger: env.logger,
+                emitPartial,
+            }
+        )
+    }
+
+    // TODO: add option to get candidate items from a specific provider.
+    const mentionsChanges = (
+        params: MentionsParams,
+        { emitPartial }: ObserveOptions,
+        providerUri?: string
+    ): Observable<EachWithProviderUri<MentionsResult>> => {
+        return observeMentions(
+            filterProviders(providerClientsWithSettings(undefined), providerUri),
+            params,
+            {
+                logger: env.logger,
+                emitPartial,
+            }
+        )
+    }
+
+    // TODO: add option to get items from a specific provider.
```

Comment:
Are these TODOs outdated? It looks like all the code around here respects the input providerUri